### PR TITLE
Deprecate MMBN Names

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
### Changelog Note
For developers with custom UIs, they can: 
* Revert #5240 and any fixes on top of it
* [Re-implement the MMBN style names](<https://github.com/rh-hideout/pokeemerald-expansion/blob/5a320f94d247310b05c8278f54c49c4cf927ac3b/src/data/items.h#L1162>) from [docs/mmbn_style_names.md](https://github.com/rh-hideout/pokeemerald-expansion/blob/76f0c90bb9ee6509fc55f0dc396ba92eae8bab91/docs/mmbn_style_names.md)
* Use `PrependFontToFit` and `FONT_SMALL_NARROWER` to get Game Freak names to work within their custom UIs

# Description
Removes `HANDLE_EXPANDED_SPECIES_NAME`, `B_EXPANDED_TYPE_NAMES`, `B_EXPANDED_ABILITY_NAMES`, `B_EXPANDED_MOVE_NAMES`, `I_EXPANDED_ITEM_NAMES`, and their associated macros.

## Details
Going forward, names will be implemented using the Official Game Freak names, building off the work from https://github.com/rh-hideout/pokeemerald-expansion/pull/4241 and https://github.com/rh-hideout/pokeemerald-expansion/pull/5101. 

# Testing
The functionality in this PR is based on the current default behavior of expansion. Further testing beyond compilation was not done.

## Clean Branch
You can recreate this branch by applying a patch or pulling the repo. From a clean version of expansion's upcoming, you can either:

### Patch
`wget https://files.catbox.moe/i5pwpq.patch -O mmbn.patch ; git apply mmbn.patch ; rm mmbn.patch`

### Repo
`git remote add psf-expansion https://github.com/PokemonSanFran/pokeemerald-expansion/ ; git pull psf-expansion deprecateMMBN`

# Discord Contact Info
I am `pkmnsnfrn` on Discord and I have started a [discussion thread](<https://discord.com/channels/419213663107416084/1268960653322883092>). 

[4241 and 5101 also have a discussion thread, for reference](https://discord.com/channels/419213663107416084/1214533910894084168/1231246595614310572)

### CODE REVIEW

File: docs/mmbn_style_names.md [hunk 1]
```diff
@@ -0,0 +1,893 @@
...
+Species | Crabminabl | Crabominable
+Species | Blacephaln | Blacephalon
+Species | Bsculegion | Basculegion
+Species | Flechinder | Fletchinder
+Type | Fight | Fighting
+Type | Electr | Electric
+Type | Psychc | Psychic
+Type | Stellr | Stellar
+
+# List
```
Comment (AsparagusEduardo) [inline docs/mmbn_style_names.md:301]: Is there a reason why this second list is here when the first one is available above?

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Just a small question in the doc file. Everything else lgtm :)

**Comment by AsparagusEduardo** [comment]
Moved changelog note to the top and renamed PR

File: docs/mmbn_style_names.md (hunk 1 above)
Comment (pkmnsnfrn) [reply docs/mmbn_style_names.md:301]: There have been a few times where I had a table and I wished it was just a list. I don't feel strongly about the form that file takes, happy to change it.

Comment (AsparagusEduardo) [reply docs/mmbn_style_names.md:301]: Please do :)
A list can be extracted by pasting the table in Excel/Google Spreadsheets and copying the respective column :)

Comment (pkmnsnfrn) [reply docs/mmbn_style_names.md]: [`ad35903` (#5240)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5240/commits/ad35903d3b2b4214f93a85983e8536d80b2b2aca)

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**